packages/frontend-core/src/components/grid/layout/ButtonColumn.svelte

```diff
@@ -0,0 +1,144 @@
+<script>
+  import { getContext, onMount } from "svelte"
+  import { Button } from "@budibase/bbui"
+  import GridCell from "../cells/GridCell.svelte"
+  import GridScrollWrapper from "./GridScrollWrapper.svelte"
+
+  const {
+    renderedRows,
+    hoveredRowId,
+    props,
+    width,
+    rows,
+    focusedRow,
+    selectedRows,
+    visibleColumns,
+    scroll,
+    isDragging,
+    buttonColumnWidth,
+  } = getContext("grid")
+
+  let measureContainer
+
+  $: buttons = $props.buttons?.slice(0, 3) || []
+  $: columnsWidth = $visibleColumns.reduce(
+    (total, col) => (total += col.width),
+    0
+  )
+  $: end = columnsWidth - 1 - $scroll.left
+  $: left = Math.min($width - $buttonColumnWidth, end)
+
+  const handleClick = async (button, row) => {
+    await button.onClick?.(rows.actions.cleanRow(row))
+    // Refresh the row in case it changed
+    await rows.actions.refreshRow(row._id)
+  }
+
+  onMount(() => {
+    const observer = new ResizeObserver(entries => {
+      const width = entries?.[0]?.contentRect?.width ?? 0
+      buttonColumnWidth.set(width)
+    })
+    observer.observe(measureContainer)
+  })
+</script>
+
+<!-- Hidden copy of buttons to measure -->
+<div class="measure" bind:this={measureContainer}>
+  <GridCell width="auto">
+    <div class="buttons">
+      {#each buttons as button}
+        <Button size="S">
+          {button.text || "Button"}
```

Comment:
Bindings aren't currently being applied to the button labels

<img width="718" alt="Screenshot 2023-11-07 at 15 49 01" src="https://github.com/Budibase/budibase/assets/5913006/934302cf-2e6b-41a9-abd6-3cf0b7c6c693">

Reply:
Nice catch! After looking into this, I think we just have to disable bindings here. The setting needs to be marked as nested, but doing so makes it think that it should be able to access the grid row context for the button name, which it actually can't. I'd also have to manually run string templates to do the enrichment, which is less than ideal as it won't properly update automatically.

So to that effect, I've basically had to just disable bindings for the text setting of buttons inside button groups, *only when the buttonConfiguration setting is nested*.

This means that in normal usage of a button group, you can bind as normal:
![image](https://github.com/Budibase/budibase/assets/9075550/7dc466ad-33b8-4a4f-a4f1-4d703fcbb019)

But when used inside a grid block for example, you cannot use bindings for the text setting:
![image](https://github.com/Budibase/budibase/assets/9075550/60aebeea-fdad-4728-acf0-a3f452eca912)

This has no impact on the other settings.